methods_and_recursion/mad_libs.rb

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env ruby
+
+# This helps to check whether 'a' or 'an' will be used.
+class String
+  def vowel?
+    vowels = %w(a e i o u A E I O U)
+    if vowels.include?(self)
+      true
+    else
+      false
+    end
+  end
+end
+
+def ask_for_word(term, list_of_words)
+  print "Give me a"
+  if list_of_words.include?(term)
+    print "nother"
+  else
+    print 'n' if term[0].vowel?
+    list_of_words << term
+  end
+  puts " #{term}:"
+  gets.strip
+end
+
+already_requested = []
+mad_libs = []
+needed_blanks = ['adjective',
+                 'adjective',
+                 'noun',
+                 'noun',
+                 'plural noun',
+                 'game',
+                 'plural noun',
+                 "verb ending in 'ing'",
+                 "verb ending in 'ing'",
+                 'plural noun',
+                 "verb ending in 'ing'",
+                 'noun',
+                 'plant',
+                 'part of the body',
+                 'place',
+                 "verb ending in 'ing'",
+                 'adjective',
+                 'number',
+                 'plural noun'
+                ]
+
+needed_blanks.each do |term|
+  mad_libs << ask_for_word(term, already_requested)
+end
+
+puts "Here is your madlib:"
```

Comment:
You can DRY up your code a bit:

    puts <<END_OF_MADLIB
      A vacation is when you take a trip to some #{mad_libs[0] place
      with your #{mad_libs[1]} family. Usually you go to some place
      ...
    END_OF_MADLIB